docs/docs/code/python/rerun/README.md

```diff
@@ -0,0 +1,188 @@
+
+```python
+def handler(pd: 'pipedream'):
+    # 7 days
+    TIMEOUT = 1000 * 60 * 60 * 24 * 7
+    pd.flow.suspend(TIMEOUT)
+  },
+})
+```
+
+## `pd.flow.rerun`
+
+Use `pd.flow.rerun` when you want to run a specific step of a workflow multiple times. This is useful when you need to start a job in an external API and poll for its completion, or have the service call back to the step and let you process the HTTP request within the step.
+
+### Polling for the status of an external job
+
+Sometimes you need to poll for the status of an external job until it completes. `pd.flow.rerun` lets you rerun a specific step multiple times:
+
+```python
+import requests
+
+def handler(pd: 'pipedream'):
+  MAX_RETRIES = 3
+  # 10 seconds
+  DELAY = 1000 * 10
+  run = pd.context['run']
+  print(pd.context)
+  # pd.context.run.runs starts at 1 and increments when the step is rerun
+  if run['runs'] == 1:
+    # pd.flow.rerun(delay, context (discussed below), max retries)
+    pd.flow.rerun(DELAY, None, MAX_RETRIES)
+  
+  elif run['runs'] == MAX_RETRIES + 1:
+    raise Exception("Max retries exceeded")
+  
+  else:
+    # Poll external API for status
+    response = requests.get("https://example.com/status")
+    # If we're done, continue with the rest of the workflow
+    if response.json().status == "DONE":
+      return response.json()
+
+    # Else retry later
+    pd.flow.rerun(DELAY, None, MAX_RETRIES)
+```
+
+`$.flow.rerun` accepts the following arguments:
+
+```python
+pd.flow.rerun(
+  delay, # The number of milliseconds until the step will be rerun
+  context, # JSON-serializable data you need to pass between runs
+  maxRetries, # The total number of times the step will rerun. Defaults to 10
+)
+```
+
+### Accept an HTTP callback from an external service
+
+When you trigger a job in an external service, and that service can send back data in an HTTP callback to Pipedream, you can process that data within the same step using `pd.flow.retry`:
+
+```python
+import requests
+
+def handler(pd: 'pipedream'):
+  TIMEOUT = 86400 * 1000
+  run = pd.context['run']
+  # pd.context['run']['runs'] starts at 1 and increments when the step is rerun
+  if run['runs'] == 1:
+    links = pd.flow.rerun(TIMEOUT, None, 1)
+    # links contains a dictionary with two entries: resume_url and cancel_url
+
+    # Send resume_url to external service
+    await request.post("your callback URL", json=links)
+  elif run['runs'] == 2:
+    raise Exception("External service never completed job")
+
+  # When the external service calls back into the resume_url, you have access to 
+  # the callback data within pd.context.run['callback_request']
+  else:
+    return run['callback_request']
+
+```
+
+### Passing `context` to `pd.flow.rerun`
+
+Within a Node.js code step, `pd.context.run.context` contains the `context` passed from the prior call to `rerun`. This lets you pass data from one run to another. For example, if you call:
+
+```python
+pd.flow.rerun(1000, { hello: "world" })
```

Comment:
I believe you'll need to wrap "hello" in double-quotes.

Reply:
I take my comment back about not having to write a lot of code in Python